### ISSUE TITLE
[Question] in which format the icons are stored in the database ?

### ISSUE THREAD

**Issue by mooms06** [keepassxreboot/keepassxc#9609]
I'm asking that question because when I manually save a favicon in the ico format, it weighs about 50 KB, but if I convert it in gif, it weighs about 2KB, so 25x smaller for no (apparent) quality loss.

My two cents: if it's not already the case, it could be a good idea to convert all the icons in an efficient image format. 

Also, it could be useful to add an export function to the icons, this way the user could verify himself the size of the icons among other things.

**Comment by droidmonkey** [comment]
Icons are stored as PNG format regardless of input format. GIF is much smaller because it is limited to 256 colors, definitely not equal quality.

**Comment by mooms06** [comment]
Thanks for the answer, I figured it myself by exporting the icons with [Custom Icon Dashboarder](https://keepass.info/plugins.html#icondashb), a Keepass plugin that allow to export the icons, and alos shows the size of the icon file. Quite handy.

I spent the last hour optimizing the icons, reducing the size from 128x128 to 64x64, then optimizing the png file with [PngOptimizer](https://psydk.org/pngoptimizer), and finally re-importing them and cleaning the database from unused icons.

My database went from 672 KB to 544 KB.
